homeassistant/components/sensor/zamg.py

```diff
@@ -62,19 +60,31 @@
 PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({
     vol.Required(CONF_MONITORED_CONDITIONS):
         vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
-    vol.Required(CONF_STATION_ID):
-        vol.All(cv.string, vol.In(VALID_STATION_IDS)),
+    vol.Optional(CONF_STATION_ID): cv.string,
```

Comment:
I can't speak authoritatively on the Austrian system, but in Australia station IDs may have leading zeros.  Without knowing much more about the schema, I left it as a string.